src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3660,24 +3662,79 @@ impl Coordinator {
                 // actually erroring here is both incredibly unlikely and
                 // recoverable (users can stop trying to drop subsources if it
                 // panics).
-                referenced_subsources.retain(|CreateSourceSubsource { subsource, .. }| {
-                    match subsource
-                        .as_ref()
-                        .unwrap_or_else(|| panic!("{}", purification_err().to_string()))
-                    {
-                        DeferredItemName::Named(name) => match name {
-                            // Retain all sources which we still have a dependency on.
-                            ResolvedItemName::Item { id, .. } => depends_on.contains(id),
+                referenced_subsources.retain(
+                    |CreateSourceSubsource {
+                         subsource,
+                         reference,
+                     }| {
+                        match subsource
+                            .as_ref()
+                            .unwrap_or_else(|| panic!("{}", purification_err().to_string()))
+                        {
+                            DeferredItemName::Named(name) => match name {
+                                // Retain all sources which we still have a dependency on.
+                                ResolvedItemName::Item { id, .. } => {
+                                    let contains = depends_on.contains(id);
+                                    if !contains {
+                                        dropped_references.insert(reference.clone());
+                                    }
+                                    contains
+                                }
+                                _ => unreachable!("{}", purification_err()),
+                            },
                             _ => unreachable!("{}", purification_err()),
-                        },
-                        _ => unreachable!("{}", purification_err()),
+                        }
+                    },
+                );
+
+                referenced_subsources.sort();
+
+                // Remove dropped references from text columns.
+                match &mut create_source_stmt.connection {
+                    CreateSourceConnection::Postgres { options, .. } => {
+                        if let Some(text_cols) = options
+                            .iter_mut()
+                            .find(|option| option.name == PgConfigOptionName::TextColumns)
+                        {
+                            match &mut text_cols.value {
+                                Some(WithOptionValue::Sequence(names)) => {
+                                    names.retain(|name| match name {
+                                        WithOptionValue::UnresolvedItemName(
+                                            column_qualified_reference,
+                                        ) => {
+                                            mz_ore::soft_assert!(
```

Comment:
The else error below is there in case of our programming error? Users should never see it?

Reply:
Correct